### PR TITLE
Add custom corne keymap

### PR DESCRIPTION
**Pull request by GauthamYerroju**
Custom corne keymap.

## Description

Second attempt at corne custom keymap with a bunch of fixes.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ X ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None

## Checklist

- [ X ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ X ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by GauthamYerroju** [comment]
Awesome, thank you!

File: keyboards/crkbd/keymaps/gotham/keymap.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT( \
+  //|-----------------------------------------------------|                    |-----------------------------------------------------|
+     KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                         KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSLS,\
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+     SFT_EQ,  SFT_A,   KC_S,    KC_D,    KC_F,    KC_G,                         KC_H,    KC_J,    KC_K,    KC_L,    SFT_SCLN,SFT_QT, \
+  //---------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+     KC_LCTL, CTL_Z,   KC_X,    KC_C,    KC_V,    KC_B,                         KC_N,    KC_M,    KC_COMM, KC_DOT,  CTL_SLSH,ADJ_GRV,\
+  //---------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                         KC_LALT, LOW_SPC, RAI_EQ,     RAI_ENT, LOW_BSP, KC_LGUI\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/gotham/keymap.c:22]:
```suggestion
  [_QWERTY] = LAYOUT(
  //|-----------------------------------------------------|                    |-----------------------------------------------------|
     KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                         KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSLS,
  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
     SFT_EQ,  SFT_A,   KC_S,    KC_D,    KC_F,    KC_G,                         KC_H,    KC_J,    KC_K,    KC_L,    SFT_SCLN,SFT_QT,
  //---------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
     KC_LCTL, CTL_Z,   KC_X,    KC_C,    KC_V,    KC_B,                         KC_N,    KC_M,    KC_COMM, KC_DOT,  CTL_SLSH,ADJ_GRV,
  //---------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
                                         KC_LALT, LOW_SPC, RAI_EQ,     RAI_ENT, LOW_BSP, KC_LGUI
```

File: keyboards/crkbd/keymaps/gotham/keymap.c [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  [_LOWER] = LAYOUT( \
+  //|-----------------------------------------------------|                    |-----------------------------------------------------|
+     KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                         KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    _______,\
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+     _______, KC_ANGL, KC_ANGR, KC_LPRN, KC_RPRN, KC_PGUP,                      KC_MINS, KC_LEFT, KC_UP,   KC_RIGHT,KC_PLUS, _______,\
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+     _______, XXXXXXX, XXXXXXX, KC_LBRC, KC_RBRC, KC_PGDOWN,                    KC_UNDS, KC_HOME, KC_DOWN, KC_END,  XXXXXXX, _______,\
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                         _______, _______, _______,    _______, LOW_DEL, _______\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/gotham/keymap.c:36]:
```suggestion
  [_LOWER] = LAYOUT(
  //|-----------------------------------------------------|                    |-----------------------------------------------------|
     KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                         KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    _______,
  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
     _______, KC_ANGL, KC_ANGR, KC_LPRN, KC_RPRN, KC_PGUP,                      KC_MINS, KC_LEFT, KC_UP,   KC_RIGHT,KC_PLUS, _______,
  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
     _______, XXXXXXX, XXXXXXX, KC_LBRC, KC_RBRC, KC_PGDOWN,                    KC_UNDS, KC_HOME, KC_DOWN, KC_END,  XXXXXXX, _______,
  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
                                         _______, _______, _______,    _______, LOW_DEL, _______
```

File: keyboards/crkbd/keymaps/gotham/keymap.c [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+  [_RAISE] = LAYOUT( \
+  //|-----------------------------------------------------|                    |-----------------------------------------------------|
+     KC_ESC,  KC_HASH, KC_DLR,  KC_LCBR, KC_RCBR, XXXXXXX,                      XXXXXXX, KC_AMPR, KC_PIPE, KC_EXLM, KC_ASTR, XXXXXXX,\
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+     _______, KC_EXLM, KC_AT,   KC_LPRN, KC_RPRN, XXXXXXX,                      XXXXXXX, KC_UNDS, KC_EQL,  KC_ANGL, KC_ANGR, XXXXXXX,\
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+     _______, KC_PERC, KC_CIRC, KC_LBRC, KC_RBRC, XXXXXXX,                      XXXXXXX, KC_MINS, KC_PLUS, XXXXXXX, KC_BSLS, _______,\
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                          _______, _______, _______,    _______, _______, _______\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/gotham/keymap.c:48]:
```suggestion
  [_RAISE] = LAYOUT(
  //|-----------------------------------------------------|                    |-----------------------------------------------------|
     KC_ESC,  KC_HASH, KC_DLR,  KC_LCBR, KC_RCBR, XXXXXXX,                      XXXXXXX, KC_AMPR, KC_PIPE, KC_EXLM, KC_ASTR, XXXXXXX,
  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
     _______, KC_EXLM, KC_AT,   KC_LPRN, KC_RPRN, XXXXXXX,                      XXXXXXX, KC_UNDS, KC_EQL,  KC_ANGL, KC_ANGR, XXXXXXX,
  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
     _______, KC_PERC, KC_CIRC, KC_LBRC, KC_RBRC, XXXXXXX,                      XXXXXXX, KC_MINS, KC_PLUS, XXXXXXX, KC_BSLS, _______,
  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
                                          _______, _______, _______,    _______, _______, _______
```

File: keyboards/crkbd/keymaps/gotham/rules.mk [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+AUDIO_ENABLE               = yes
+UNICODE_ENABLE             = no
+BLUETOOTH_ENABLE           = no
+RGBLIGHT_ENABLE            = yes
+SWAP_HANDS_ENABLE          = no
+RGB_MATRIX_ENABLE          = no
+HAPTIC_ENABLE              = no
+SLEEP_LED_ENABLE           = no
+OLED_DRIVER_ENABLE         = yes
+LINK_TIME_OPTIMIZATION_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/gotham/rules.mk:22]:
```suggestion
MOUSEKEY_ENABLE = no
EXTRAKEY_ENABLE = yes
AUDIO_ENABLE = yes
RGBLIGHT_ENABLE = yes
OLED_DRIVER_ENABLE = yes
LTO_ENABLE = yes
```

**Comment by GauthamYerroju** [comment]
Done. Also pulled and tested suggested changes.

**Comment by noroadsleft** [comment]
Thanks!

**Comment by GauthamYerroju** [comment]
Thank you all!

**Comment by drashna** [comment]
Since I don't know how to contact you otherwise, in your readme, you mention issues with getting the oled on the secondary/slave side to turn off.  I actually have a fix for that, but for now, it requires a custom transport. 

https://github.com/drashna/qmk_firmware/blob/drashna_keymaps/users/drashna/drashna_transport.c#L306

https://github.com/drashna/qmk_firmware/blob/drashna_keymaps/users/drashna/drashna_transport.c#L437

https://github.com/drashna/qmk_firmware/blob/drashna_keymaps/users/drashna/drashna_transport.c#L488-L494

You also need to disable timeout, and handle it on the master side, manually. 

https://github.com/drashna/qmk_firmware/blob/drashna_keymaps/users/drashna/oled_stuff.c#L344-L354

I can verify that this works perfectly.  The secondary OLED turns on and off in sync with the main side.  

Also, I have a bunch of additional info pushed over the transport, and can display a bunch of info on it (such as mod state, layers, etc)

**Comment by GauthamYerroju** [comment]
Hey Drashna,

Thanks! Irs been a while since I worked on QMK, but I'll see if I can get
that fixed in my next PR.

Best,

**Comment by drashna** [comment]
You're very welcome.  I figured it was worth mentioning regardless

**Pull request merged**